service/raft/group0_state_machine.cc

```diff
@@ -169,7 +169,8 @@ future<> group0_state_machine::merge_and_apply(group0_state_machine_merger& merg
         co_await _ss.topology_transition();
     },
     [&] (service_levels_change& chng) -> future<> {
-        return write_mutations_to_database(_sp, cmd.creator_addr, std::move(chng.mutations));
+        co_await write_mutations_to_database(_sp, cmd.creator_addr, std::move(chng.mutations));
+        co_await _ss.update_service_levels_configuration();
```

Comment:
I think a different order of the commits would make more sense: first introduce `update_service_levels_configuration`, then introduce `service_levels_change`. This way you will be able to write a correct handler for `service_levels_change` at once.